### PR TITLE
WIP: Room dependency migration to convention plugin

### PR DESCRIPTION
**Pull request by misshannah**
# Scope

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [ ] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
Closes one section of issue 33

## Other testing QA Notes
Working on this with @jumaallan 

Please add a screenshot (if necessary)
No screenshot required, just confirmed the app is running on Android studio.

### CODE REVIEW

**Review by kibettheophilus — APPROVED** [review]

File: data/build.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,7 @@
 plugins {
     id("droidconke.android.library")
     id("com.google.devtools.ksp")
```
Comment (jumaallan) [inline data/build.gradle.kts:18]: You can now remove line 18, as the room plugin is providing ksp to this module

Comment (jumaallan) [reply data/build.gradle.kts:18]: On the same file, remove this block as well

```
 ksp {
        arg("room.schemaLocation", "$projectDir/schemas")
    }
```

Its also configured inside `AndroidRoomConventionPlugin`

**Review by jumaallan — CHANGES_REQUESTED** [review]
Left 2 small comments

Fix those and this should be ready to merge 🥳

File: data/build.gradle.kts (hunk 1 above)
Comment (misshannah) [reply data/build.gradle.kts:18]: Thanks, I get it now.

**Review by jumaallan — APPROVED** [review]

**Review by wangerekaharun — APPROVED** [review]

**Pull request merged**